core/src/main/java/edu/illinois/cs/cogcomp/l2ilp/inference/CCMLogicSolver.java

```diff
@@ -38,21 +38,23 @@
     public CCMLogicSolver(
         List<Pair<BooleanVariable, Double>> objective,
         List<LogicFormula> hardConstraints,
-        List<Pair<LogicFormula, Double>> softConstraints, boolean maximizeObjective) {
+        List<Pair<LogicFormula, Double>> softConstraints,
+        boolean maximizeObjective,
+        ILPProblem problem) {
         this.objective = objective;
         this.hardConstraints = hardConstraints;
         this.softConstraints = softConstraints;
         this.variableCounter = new Counter("NV$");
         this.constraintCounter = new Counter("C$");
-        this.problem = null;
+        prepare(problem);
```

Comment:
Change the order of 49 and 50.
Otherwise prepare will access this.maximizeObjective, which is undefined at this point.